### PR TITLE
Allow open delegates to nullables

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Port of https://github.com/dotnet/runtime/pull/42837.

Won't win any perf prizes, but will do for now. I don't think the amount of work required for good support will ever meet the bar.

### CODE REVIEW

File: src/coreclr/nativeaot/System.Private.Reflection.Core/src/System/Reflection/Runtime/MethodInfos/CustomMethodInvoker.cs [hunk 1]
```diff
@@ -53,8 +53,30 @@ public sealed override Delegate CreateDelegate(RuntimeTypeHandle delegateType, o
         {
             if (_thisType.IsConstructedGenericType && _thisType.GetGenericTypeDefinition() == typeof(Nullable<>))
             {
-                // Desktop compat: MethodInfos to Nullable<T> methods cannot be turned into delegates.
-                throw new ArgumentException(SR.Arg_DlgtTargMeth);
+                if (isOpen)
+                {
+                    return DynamicDelegateAugments.CreateObjectArrayDelegate(Type.GetTypeFromHandle(delegateType),
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.Reflection.Core/src/System/Reflection/Runtime/MethodInfos/CustomMethodInvoker.cs:58]: This won't work for methods that take refs, byref-like types, etc. Is that right?

**Comment by jkotas** [comment]
> Won't win any perf prizes, but will do for now.

I agree that it makes the test pass, but I am not sure whether partial slow implementation like this is an improvement. It may be better to just keep it broken and wait until somebody actually needs it.

File: src/coreclr/nativeaot/System.Private.Reflection.Core/src/System/Reflection/Runtime/MethodInfos/CustomMethodInvoker.cs (hunk 1 above)
Comment (MichalStrehovsky) [reply src/coreclr/nativeaot/System.Private.Reflection.Core/src/System/Reflection/Runtime/MethodInfos/CustomMethodInvoker.cs:58]: Yes, but Nullable doesn't have any of those. This only works for the following methods:

https://github.com/dotnet/corert/blob/24f1edc938af9954eac9a3385958787ca64956d5/src/System.Private.Reflection.Core/src/System/Reflection/Runtime/MethodInfos/CustomMethodMapper.Nullable.cs#L28-L96

**Comment by MichalStrehovsky** [comment]
>  It may be better to just keep it broken and wait until somebody actually needs it.

That will just mean we'll have a bug open and an extra ActiveIssue that we will have roll around from repo to repo. It felt like adding localized code that we will never have to touch will save us time taking care and feeding a bug forever.

File: src/coreclr/nativeaot/System.Private.Reflection.Core/src/System/Reflection/Runtime/MethodInfos/CustomMethodInvoker.cs (hunk 1 above)
Comment (jkotas) [reply src/coreclr/nativeaot/System.Private.Reflection.Core/src/System/Reflection/Runtime/MethodInfos/CustomMethodInvoker.cs:58]: ah ok, I thought that it is for more than that.

**Review by jkotas — APPROVED** [review]

**Pull request merged**